x-pack/plugins/monitoring_collection/server/lib/verify_mappings.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { get } from 'lodash';
+import { ElasticsearchClient, Logger } from 'kibana/server';
+import { MakeSchemaFrom } from '../../../../../src/plugins/usage_collection/server';
+import { Metric } from '../plugin';
+
+interface Params {
+  client: ElasticsearchClient;
+  name: string;
+  metrics: Array<Metric<any>>;
+  logger: Logger;
+}
+
+function buildSchemaFields(schema: MakeSchemaFrom<any>, paths: string[] = []) {
+  return Object.entries(schema).reduce((accum: string[], [key, value]) => {
+    if (typeof value.type === 'string') {
+      accum.push([...paths, key].join('.properties.'));
+    } else {
+      accum.push(...buildSchemaFields(value as MakeSchemaFrom<any>, [...paths, key]));
+    }
+    return accum;
+  }, []);
+}
+
+export async function verifyMappings({ client, name, metrics, logger }: Params) {
+  const typesToDisable = new Set<string>();
+  try {
+    const [{ body: templates }, { body: template }] = await Promise.all([
+      client.indices.getIndexTemplate({ name: `${name}*` }),
+      client.indices.getTemplate({ name: `${name}*` }),
```

Comment:
We can remove the call to the legacy `getTemplate` and all references to `template` as we won't support internal monitoring

Reply:
This file is moot now that we only will support MB so it will be removed